R/zarr.R

```diff
@@ -0,0 +1,248 @@
+    initialize = function() {
+      self$root <- obj_list()
+    },
+    #' @description
+    #' Get the parent of an item.
+    #' @keywords internal
+    #' @param item The item key.
+    #' @return A list with the keys `parent` and `key`.
+    get_parent = function(item) {
+      parent <- self$root
+      segments <- strsplit(item, "/")[[1]]
+      for(k in segments[1:length(segments)-1]) {
+        parent <- parent[[k]]
+        if(!is.list(parent)) {
+          stop("KeyError:", item)
+        }
+      }
+      return(list(parent = parent, key = segments[length(segments)]))
+    },
+    #' @description
+    #' Get an item from the store.
+    #' @param item The item key.
+    #' @return The item data in a vector of type raw.
+    get_item = function(item) {
+      parent_and_key <- self$get_parent(item)
+      parent <- parent_and_key$parent
+      key <- parent_and_key$key
+
+      if(key %in% names(parent)) {
+        value <- parent[[key]]
+      } else {
+        stop("KeyError:", item)
+      }
+      return(value)
+    },
+    #' @description
+    #' Set an item in the store.
+    #' @param item The item key.
+    #' @param value The item value as a vector of type raw.
+    set_item = function(item, value) {
+      segments <- strsplit(item, "/")[[1]]
+      if(length(segments) > 1) {
+        for(i in 1:(length(segments)-1)) {
+          k <- segments[i]
+          if(i == 1 && k %in% names(self$root)) {
+            if(!is.list(self$root[[k]])) {
+              stop("KeyError:", item)
+            }
+          } else if(i > 1 && k %in% names(self$root[[segments[1:(i-1)]]])) {
+            if(!is.list(self$root[[segments[1:i]]])) {
+              stop("KeyError:", item)
+            }
+          } else {
+            self$root[[segments[1:i]]] <- obj_list()
+          }
+        }
+      }
+      self$root[[segments]] <- value
+    },
+    #' @description
+    #' Determine whether the store contains an item.
+    #' @param item The item key.
+    #' @return A boolean value.
+    contains_item = function(item) {
+      result <- tryCatch({
+        parent_and_key <- self$get_parent(item)
+        parent <- parent_and_key$parent
+        key <- parent_and_key$key
+        if(key %in% names(parent)) {
+          value <- parent[[key]]
+          return(!is.list(value))
+        } else {
+          return(FALSE)
+        }
+      }, error = function(e) {
+        return(FALSE)
+      })
+      return(result)
+    }
+  )
+)
+
+
+#' Convert a JSON-like list to a raw type.
+#'
+#' @param json_as_list An R list to be converted to JSON.
+#' @return The raw value.
+json_to_raw <- function(json_as_list) {
+  json_str <- jsonlite::toJSON(json_as_list)
+  json_raw <- charToRaw(json_str)
+  return(json_raw)
+}
+
+
+#' Write an R matrix to a Zarr store (one chunk, no compression).
+#'
+#' @param matrix The matrix as an R matrix.
+#' @param rows A vector of row names.
+#' @param cols A vector of column names.
+#' @param store The Zarr store.
```

Comment:
Why no `compressor` param?

Reply:
Also why isn't the default compressor the base `Codec` class?  It seems like it does what we need.  Could get rid of the check [here](https://github.com/vitessce/vitessce-r/pull/17/files#diff-f8c70034960c68753d41e7c231561f5a28876b04fa139bc2d083accd72655423R223-R226)